### PR TITLE
Add pyright to Travis CI

### LINKED ISSUE #23: Configure and Integrate microsoft/pyright in the CI to enforce static typing
**Requirements:**
 - Configure & Integrate this - https://github.com/microsoft/pyright
 - Add guidelines in the README to run it on the local. 
 - Travis Integration to validate code changes in the pull requests.

### PR DESCRIPTION
**Pull request by yankeexe**
resolves #23

* Added checks to: 
    - reportMissingImports 
    - reportUnusedImport
    - reportUnusedFunction 
    - reportUnusedVariable
    - reportDuplicateImport 

* Added type stubs to check for `pyodbc` related class and method types.
* Added `executionEnvironments` to address local import resolution.

### CODE REVIEW

File: pyrightconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "include": ["examples/**/*.py", "test/**/*.py"],
+    "exclude":["venv", "*.pyc", "*.egg-info", ".coverage"],
+
+    "reportMissingImports": true,
+    "reportUnusedImport": true,
+    "reportUnusedFunction": true,
+    "reportUnusedVariable": true,
+    "reportDuplicateImport": true
+}
```
Comment (mesaugat) [inline pyrightconfig.json]: We start again? @yankeexe 

Line feed at the end.

File: pyrightconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "include": ["examples/**/*.py", "test/**/*.py"],
+    "exclude":["venv", "*.pyc", "*.egg-info", ".coverage"],
+
```
Comment (mesaugat) [inline pyrightconfig.json]: Empty line for no reason?

Comment (yankeexe) [reply pyrightconfig.json]: This is to separate out the include, exclude keys with the rest of the checks.

Comment (mesaugat) [reply pyrightconfig.json]: No need to do that.

**Comment by kabirbaidhya** [comment]
@yankeexe Fix the build. It's failing.

**Comment by yankeexe** [comment]
> @yankeexe Fix the build. It's failing.

@mesaugat @kabirbaidhya dai can you guys check if things are proper now?

File: pyrightconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "include": ["examples/**/*.py", "test/**/*.py"],
+    "exclude":["venv", "*.pyc", "*.egg-info", ".coverage"],
```
Comment (mesaugat) [inline pyrightconfig.json]:
```suggestion
    "exclude": ["venv", "*.pyc", "*.egg-info", ".coverage"],
```

**Review by mesaugat — APPROVED** [review]
Looks good @yankeexe 👍

**Review by kabirbaidhya — APPROVED** [review]

**Review by kabirbaidhya — APPROVED** [review]

**Pull request merged**